command.go

```diff
@@ -119,6 +119,35 @@ func cmdString(cmd Cmder, val interface{}) string {
 	return util.BytesToString(b)
 }
 
+// A wrapper around extract keys, just to make it easier to test.
+func GetKeys(cmd Cmder) []string {
+	return extractKeys(cmd)
+}
+
+func extractKeys(cmd Cmder) []string {
+	firstKeyPos := cmdFirstKeyPos(cmd)
+	if firstKeyPos == -1 {
+		return nil
+	}
+
+	args := cmd.Args()
+	keys := []string{}
+
+	switch cmd.Name() {
+	case "mget":
```

Comment:
This approach isn't scalable, each time we need a new type of extracting we have to add a command here. `Cmder` is an interface that should be abstract as possible and should not contain any implementation specific (mget is an implementation specific in this case).

Cmder should return a keys, but shouldn't be aware about the keys position of specific command (this is implementation details), instead the specific command that implements this interface should define keys position for itself.